pkg/eestream/decode.go

```diff
@@ -285,31 +285,36 @@ func (dr *decodedRanger) Range(offset, length int64) io.ReadCloser {
 	// do it parallel to save from network latency
 	readers := make(map[int]io.ReadCloser, len(dr.rrs))
 	type indexReadCloser struct {
-		i int
-		r io.ReadCloser
+		i   int
+		r   io.ReadCloser
+		err error
 	}
 	result := make(chan indexReadCloser, len(dr.rrs))
 	for i, rr := range dr.rrs {
 		go func(i int, rr ranger.Ranger) {
-			r := rr.Range(
+			r, err := rr.Range(
 				firstBlock*int64(dr.es.EncodedBlockSize()),
 				blockCount*int64(dr.es.EncodedBlockSize()))
-			result <- indexReadCloser{i, r}
+			result <- indexReadCloser{i, r, err}
 		}(i, rr)
 	}
 	// wait for all goroutines to finish and save result in readers map
 	for range dr.rrs {
 		res := <-result
-		readers[res.i] = res.r
+		if res.err != nil {
+			readers[res.i] = readcloser.FatalReadCloser(res.err)
```

Comment:
@jtolds Please check this closer. Here I still use FatalReadCloser because I don't want to fail the `decodedReader.Range` due to failed `Range` for some pieces.

Reply:
seems like a good approach to me